cloud/amazon/cloudformation_assemble.py

```diff
@@ -0,0 +1,149 @@
+
+import sys
+import yaml
+import json
+
+DOCUMENTATION = '''
+---
+module: cloudformation_assemble
+short_description: assemble an AWS Cloudformation JSON template from a collection of YML files
+description:
+     - Convert YML to JSON.
+     - Combine many files so a single template can be sanely organized
+version_added: "2.0"
+options:
+  directory:
+    description:
+      - base directory for this cloudformation stack configuration
+    default: "cfbuild"
+  destination:
+    description:
+      - file where compiled template JSON will be placed
+    default: "/tmp/cfbuild.json"
+  subdirs:
+    description:
+      - if absent, all subdirectories will be compiled
+      - if present, comma-delimited list of subdirectories that will be compiled
+
+notes:
+  - The file structure for cloudformation_assemble expects subdirectories to be created. Inside those are up to five yml files, corresponding to the five sections of a Cloudformation template. Here's a sample layout:
+    cfbuild/meta.yml
+    cfbuild/web/resources.yml
+    cfbuild/web/mappings.yml
+    cfbuild/web/outputs.yml
+    cfbuild/web/conditions.yml
+    cfbuild/web/parameters.yml
+    cfbuild/web/resources.yml
+
+requirements: [ "boto" ]
+author: Ted Timmons, Andrew Cholakian
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Assuming the following files are in place:
+mkdir -p files/cf-example/basic-server/
+cat <<"EOF" > files/cf-example/basic-server/resources.yml
+server:
+  Type: AWS::EC2::Instance
+  Properties:
+    KeyName: "my-ssh-key"
+    ImageId: ami-d85e75b0
+    InstanceType: t1.micro
+    AvailabilityZone: "us-east-1a"
+EOF
+
+cat <<"EOF" > files/cf-example/meta.yml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Sample Stack
+EOF
+
+## simple example
+
+
+- name: assemble subset of a cloudformation stack
+  local_action: cloudformation_assemble directory=files/cf-example subdirs=basic-server
+  run_once: true
+- name: Launch cloudformation using compiled json
+  local_action: cloudformation stack_name="test1" state=present region=us-east-1 disable_rollback=yes template=/tmp/cfbuild.json
+  run_once: true
+
+## more typical example, uses files in http://github.com/stansonhealth
+
+- name: assemble entire cloudformation stack
+  local_action: cloudformation_assemble directory=files/cf-typical
+  run_once: true
+- name: Launch cloudformation using compiled json
+  local_action: cloudformation stack_name="test2" state=present region=us-east-1 disable_rollback=yes template=/tmp/cfbuild.json
+  run_once: true
+  args:
+    template_parameters:
+      KeyName: "{{aws_east_key_name}}"
+'''
+
+def read_yaml(filepath):
+    f = open(filepath, 'r')
+    if f:
+        return yaml.load(f) or dict()
+    else:
+        raise Exception("Could not find file: " + f)
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            directory=dict(default="cfbuild"),
+            destination=dict(default="/tmp/cfbuild.json"),
```

Comment:
I don't agree with having a default for this parameter. Also, may need to be specified as required?